source/Plugins/Language/Swift/SwiftFormatters.cpp

```diff
@@ -242,6 +242,32 @@ bool lldb_private::formatters::swift::StringGuts_SummaryProvider(
     if (objectAddr == LLDB_INVALID_ADDRESS)
       return false;
 
+    uint64_t topNibbleMask = 0xF000000000000000;
+    uint64_t smallUTF8TopNibble = 0xE000000000000000;
+
+    // Small UTF-8 string
+    if ((objectAddr & topNibbleMask) == smallUTF8TopNibble) {
+      // Small UTF-8 strings store their contents directly in register. The
+      // high nibble of the most significant byte of the first word denotes
+      // that the string is small, the low nibble holds the count. The least-
+      // significant byte of the second word is the first UTF-8 code unit.
+      //
+      // {object, otherBits} =
+      //   { isSmallUTF8Nibble << 60 | count << 56 | c_14 << 48 | ... | c_8,
+      //     c_7 << 56 | c_6 << 48 | ... | c_0
+      //   }
+      //
+      //
+      uint64_t countMask = 0x0F00000000000000;
+      uint64_t count = (objectAddr & countMask) >> 56U;
+      assert(count <= 15 && "malformed small string");
+      uint64_t buffer[2] = {otherBits->GetValueAsUnsigned(0), objectAddr};
+      stream.Printf("\"");
+      stream.Write(&buffer, count);
```

Comment:
Because 0 isn't a terminator, you'll need to explicitly escape it here in order for it to render properly. I suggest using ReadBufferAndDumpToStreamOptions + DumpUTFBufferToStream to do this for you.

Reply:
So I spent some time trying to understand StringPrinter, figured out that was a waste of time and tried to copy the approach that's done for small Character. Here's what I wrote, however it gives me the same results as before with '\0' in the middle of a string:

```c++
      // Small UTF-8 strings store their contents directly in register. The
      // high nibble of the most significant byte of the first word denotes
      // that the string is small, the low nibble holds the count. The least-
      // significant byte of the second word is the first UTF-8 code unit.
      //
      // {object, otherBits} =
      //   { isSmallUTF8Nibble << 60 | count << 56 | c_14 << 48 | ... | c_8,
      //     c_7 << 56 | c_6 << 48 | ... | c_0
      //   }
      //
      //
      uint64_t countMask = 0x0F00000000000000;
      uint64_t count = (objectAddr & countMask) >> 56U;
      assert(count <= 15 && "malformed small string");
      uint64_t buffer[2] = {otherBits->GetValueAsUnsigned(0), objectAddr};

      DataExtractor data(buffer, count, process->GetByteOrder(),
                         process->GetAddressByteSize());

      StringPrinter::ReadBufferAndDumpToStreamOptions options;
      options.SetData(data)
          .SetEscapeNonPrintables(true)
          .SetPrefixToken(0)
          .SetQuote('"')
          .SetStream(&stream)
          .SetSourceSize(count)
          .SetLanguage(lldb::eLanguageTypeSwift);
      return StringPrinter::ReadBufferAndDumpToStream<
          StringPrinter::StringElementType::UTF8>(options);
```

Any ideas?

Reply:
Argh, I figured out that I need `.SetBinaryZeroIsTerminator(false)`, which is completely confounding as what is the point of a length, and why doesn't small Character have to do this?

Reply:
And why is StringPrinter highly unidiomatic and completely devoid of comments?